code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm

```diff
@@ -578,11 +578,336 @@
+/datum/status_effect/seedling_beam_indicator/on_apply()
+	beam_overlay = mutable_appearance('icons/mob/jungle/arachnid.dmi', "seedling_beam_indicator")
+	beam_overlay.pixel_x = -16 -owner.pixel_x
+	beam_overlay.pixel_y = -16 -owner.pixel_y
+	tick()
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/Destroy()
+	if(owner)
+		owner.overlays -= beam_overlay
+	QDEL_NULL(beam_overlay)
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/tick()
+	var/target_angle = Get_Angle(owner, target)
+	owner.overlays -= beam_overlay
+	angle = target_angle
+	var/matrix/final = matrix()
+	final.Turn(target_angle)
+	beam_overlay.transform = final
+	owner.overlays += beam_overlay
+
+
+/datum/status_effect/seedling_beam_indicator/be_replaced()
+	owner.underlays -= beam_overlay
+	..()
+
+/mob/living/simple_animal/hostile/jungle/seedling/Initialize()
+	. = ..()
+	var/obj/seedling_weakpoint/W = new /obj/seedling_weakpoint(src)
+	weak_point = W
+	W.our_seedling = src
+
+/mob/living/simple_animal/hostile/jungle/seedling/Goto()
+	if(combatant_state != SEEDLING_STATE_NEUTRAL)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/jungle/seedling/AttackingTarget()
+	if(isliving(target))
+		if(ranged_cooldown <= world.time && combatant_state == SEEDLING_STATE_NEUTRAL)
+			OpenFire(target)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/jungle/seedling/OpenFire()
+	WarmupAttack()
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/WarmupAttack()
+	if(combatant_state == SEEDLING_STATE_NEUTRAL)
+		combatant_state = SEEDLING_STATE_WARMUP
+		walk(src,0)
+		update_icons()
+		var/target_dist = get_dist(src,target)
+		var/living_target_check = isliving(target)
+		if(living_target_check)
+			if(target_dist > 7)//Offscreen check
+				SolarBeamStartup(target)
+				return
+			if(get_dist(src,target) >= 4 && prob(40))
+				SolarBeamStartup(target)
+				return
+		addtimer(CALLBACK(src, .proc/Volley), 5)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/SolarBeamStartup(mob/living/living_target)//It's more like requiem than final spark
+	if(combatant_state == SEEDLING_STATE_WARMUP && target)
+		combatant_state = SEEDLING_STATE_ACTIVE
+		living_target.apply_status_effect(/datum/status_effect/seedling_beam_indicator, src)
+		beam_debuff_target = living_target
+		playsound(src,'sound/effects/seedling_chargeup.ogg', 100, 0)
+		if(get_dist(src,living_target) > 7)
+			playsound(living_target,'sound/effects/seedling_chargeup.ogg', 100, 0)
+		solar_beam_identifier = world.time
+		addtimer(CALLBACK(src, .proc/Beamu, living_target, solar_beam_identifier), 35)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/Beamu(mob/living/living_target, beam_id = 0)
+	if(combatant_state == SEEDLING_STATE_ACTIVE && living_target && beam_id == solar_beam_identifier)
+		if(living_target.z == z)
+			update_icons()
+			solar_beam = new(src,target,time=5,beam_icon_state="solar_beam",maxdistance = INFINITY,btype=/obj/effect/ebeam/solarbeam,beam_sleep_time = 0.5)
```

Comment:
I recommend making this from the turf to the target, since beam handles larger icon files weirdly.
You can see how in the gif.